br/pkg/restore/split_client.go

```diff
@@ -114,6 +116,24 @@ func (c *pdClient) needScatter(ctx context.Context) bool {
 	return c.needScatterVal
 }
 
+// ScatterRegions scatters regions in a batch.
+func (c *pdClient) ScatterRegions(ctx context.Context, regionInfo []*RegionInfo) error {
+	c.mu.Lock()
```

Comment:
Confused a little. What does Mutex protect?
if protect c.client.ScatterRegions(ctx, regionsID), move c.mu.Lock() to before line:127
if protect resp.GetHeader(), move lock() to before line:131
if no use, remove it ?
Suggest~

Reply:
I guess it can be removed, seems it was guard for `c.storeCache`. 🤔

Reply:
Thankfully, this extra mutex won't make things wrong, because the c.storeCache is accessed at the vary start of restoring.